src/Containers.jsx

```diff
@@ -617,64 +663,71 @@ class Containers extends React.Component {
+                // FIXME: what about --user and system with the same unit name, make it a dict?
+                if (service_name) {
+                    running_pod_quadlets.add(utils.makeKey(pod.uid, service_name));
+                    podServiceNameIdMap[service_name] = utils.makeKey(pod.uid, pod.Id);
+                }
+            });
 
-            filtered.sort((a, b) => {
-                // Show unhealthy containers first
-                const a_health = getHealth(a);
-                const b_health = getHealth(b);
-                if (a_health !== b_health) {
-                    if (a_health === "unhealthy")
-                        return -1;
-                    if (b_health === "unhealthy")
-                        return 1;
+            Object.keys(this.props.quadletPods).forEach(key => {
+                const values = this.props.quadletPods[key];
+                if (!running_pod_quadlets.has(utils.makeKey(values.uid, values.Labels.PODMAN_SYSTEMD_UNIT))) {
+                    pods[key] = values;
                 }
-                // User containers are in front of system ones
-                if (this.props.containers[a].uid !== this.props.containers[b].uid)
-                    return (this.props.containers[a].uid === 0) ? 1 : -1;
-                return this.props.containers[a].Name > this.props.containers[b].Name ? 1 : -1;
             });
 
-            Object.keys(this.props.pods || {}).forEach(pod => { partitionedContainers[pod] = [] });
+            Object.keys(pods).forEach(pod => { partitionedContainers[pod] = [] });
 
+            const running_quadlets = new Set();
             filtered.forEach(id => {
                 const container = this.props.containers[id];
-                if (container)
+                if (container) {
                     (partitionedContainers[container.Pod ? utils.makeKey(container.uid, container.Pod) : 'no-pod'] || []).push(container);
+                    const service_name = container?.Config?.Labels?.PODMAN_SYSTEMD_UNIT;
+                    // FIXME: what about --user and system with the same unit name, make it a dict?
+                    if (service_name)
+                        running_quadlets.add(utils.makeKey(container.uid, service_name));
+                }
+            });
+
+            const filteredQuadlets = this.filterContainers(this.props.quadletContainers);
+            filteredQuadlets.forEach(id => {
+                if (!running_quadlets.has(id)) {
+                    const cont = this.props.quadletContainers[id];
+                    if (cont.Pod && !running_pod_quadlets.has(cont.Pod)) {
+                        const podKey = utils.makeKey(cont.uid, cont.Pod);
+                        // Containers and pod state aren't updated in sync
+                        if (podKey in partitionedContainers)
+                            partitionedContainers[podKey].push(cont);
+                    } else if (cont.Pod && podServiceNameIdMap[cont.Pod]) {
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2266-91a6840b-20250918-095040-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)